build.gradle

```diff
@@ -111,7 +111,9 @@ subprojects {
   targetCompatibility = '1.8'
 
   dependencies {
-    implementation 'org.slf4j:slf4j-api'
+    if (!(project.name.contains("spark") && project.name.contains("runtime"))) {
+      implementation 'org.slf4j:slf4j-api'
+    }
```

Comment:
I don't think this is the right way to do this. We want to only change the iceberg-spark-runtime-3.2_2.12 module, not all Spark runtime modules. The others work fine so there's no need to change them.

I think we have 2 options. First, we can move this implementation dependency into every module. That may be the most straightforward. Second, I think we could probably suppress this in just the iceberg-spark-runtime-3.2_2.12 module using something like this:

```
configurations.implementation {
    exclude group: 'org.slf4j', module: 'slf4j-api'
}
```

Reply:
I'm not sure if other spark versions work fine, I'll check and update here

Reply:
actually the same issue occurs for all spark versions (2.4, 3.0, 3.1 and 3.2). That's why I'm changing all spark runtime modules

Reply:
I tried to test Spark 3.1 and the module appears to be missing from nightlies. I'll look into that...

Reply:
The nightlies are missing because we only build with the defaults. I'll build locally to look into this more.

Reply:
Yep, I ran into that too. We probably should add the `-DflinkVersions=1.12,1.13,1.14 -DsparkVersions=2.4,3.0,3.1,3.2 -DhiveVersions=2,3` properties to the publish snapshot workflow. Right now we're publishing only the default versions (for spark it's `systemProp.defaultSparkVersions=3.2` set in gradle.properties)

Reply:
I can open another PR adding this, if it makes sense